lib/pure/algorithm.nim

```diff
@@ -8,19 +8,40 @@
 #
 
 ## This module implements some common generic algorithms.
+##
+## .. code-block::
+##    import algorithm
+##
+##    var birth = ["2000 John", "1995 Marie", "2010 Jane"]
+##    sort(birth)
+##    assert birth == @["1995 Marie", "2000 John", "2010 Jane"]
+##    echo "Firstborn: " & birth[0] # --> 1995 Marie
+##
+##    let john = binarySearch(birth, "2000 John")
+##    echo "John was born as number " & $(john + 1) # --> 2
+##
+##
+## See also:
```

Comment:
If there are no other headings before it (e.g. "Basic usage"), then I didn't use heading for this, just `**See also:**`. (And when I turn this into a heading, then I delete the `:`, heading looks better without it, IMO)